storage/buckets/main.go

```diff
@@ -197,3 +198,71 @@ func removeUser(c *storage.Client, bucketName string) error {
 	// [END remove_bucket_iam_member]
 	return nil
 }
+
+func enableRequesterPays(c *storage.Client, bucketName string) error {
+	ctx := context.Background()
+
+	// [START enable_requester_pays]
+	bucket := c.Bucket(bucketName)
+	if _, err := bucket.Update(ctx, storage.BucketAttrsToUpdate{
+		RequesterPays: true,
+	}); err != nil {
+		return err
+	}
+	// [END enable_requester_pays]
+	return nil
+}
+
+func disableRequesterPays(c *storage.Client, bucketName string) error {
+	ctx := context.Background()
+
+	// [START disable_requester_pays]
+	bucket := c.Bucket(bucketName)
+	if _, err := bucket.Update(ctx, storage.BucketAttrsToUpdate{
+		RequesterPays: false,
+	}); err != nil {
+		return err
+	}
+	// [END disable_requester_pays]
+	return nil
+}
+
+func checkRequesterPays(c *storage.Client, bucketName string) error {
+	ctx := context.Background()
+
+	// [START get_requester_pays_status]
+	attrs, err := c.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Is requester pays enabled? %v\n", attrs.RequesterPays)
+	// [END get_requester_pays_status]
+	return nil
+}
+
+func downloadUsingRequesterPays(client *storage.Client, object, bucketName, localpath, billingProjectID string) error {
+	ctx := context.Background()
+	// [START storage_download_file_requester_pays]
+	bucket := client.Bucket(bucketName).UserProject(billingProjectID)
+	src := bucket.Object(object)
+
+	f, err := os.OpenFile(localpath, os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
+		return err
+	}
+	rc, err := src.NewReader(ctx)
+	if err != nil {
+		return err
+	}
+	if _, err := io.Copy(f, rc); err != nil {
+		return err
+	}
+	if err := rc.Close(); err != nil {
+		return err
+	}
+	fmt.Printf("Copied file.txt using %v as billing project.\n", billingProjectID)
```

Comment:
I think the printf message should be updated as the file is downloaded instead of copied.

One other follow-up: Given the organization of current samples, should the download sample be in the objects sample?

Reply:
Done.

I moved it.